### DISCUSSION TITLE
A slimmer fork of networkx?

### DISCUSSION THREAD

**Discussion by FlickerSoul** [networkx/networkx#5113] [Q&A] [answered]
Hey! 

I'm working on integrating networkx into a tutorial module that teaches intro graph algorithm but networkx is way too big for that mater. Is there a fork or somethings that's slimmer and only carries basic graph supports? 

Best regards,
L

**Comment by dschult** [comment]
Why not have the tutorial install networkx and then they can just import it when they need it?
There is no reason to copy and include a separate copy of networkx for your tutorial.
And the students should learn how to install packages anyway...

**Reply by FlickerSoul** [reply]
Thank you so much for the suggestion! I'm going to write some wrappers for some classes in networkx so a subset of the module is easier to work with and ship. Do you happen to know any candidates?

**Comment by dschult** [comment, marked as answer]
Perhaps the best solution is just using Python's dict.  
```python
G = {n1: [n2, n3, n4], n2: [n1], n3: [n1], n4: [n1]}
for n, nbrs in G.items():
    for nbr in nbrs:
        print(f"edge: {n} -> {nbr}")
```

**Reply by FlickerSoul** [reply]
Thank you so much!
